### PR TITLE
Summaries subdialog

### LINKED ISSUE #4497: Summaries - first and last and nth
The summaries dialogue (and hence climatic summaries has changed a lot. It is looking very impressive now. 

We now have First and Last and nth. What remains is the add the **OrderBy** facility to these summaries. I suggest we keep First/Last/nth where they are now. 

In addition we repeat them in the 2-variable tab - with the OrderBy label for that block of summaries. So it is effectively another statistic - I hope that is OK?

### PR DESCRIPTION
**Pull request by shadrackkibet**
@africanmathsinitiative/developers  this is ready for review. Fixes #4497

### CODE REVIEW

**Comment by shadrackkibet** [comment]
@africanmathsinitiative/developers someone should test this!

**Comment by Muthenya** [comment]
This looks good from my end. @dannyparsons this is ready for you.

File: instat/sdgSummaries.vb [hunk 1]
```diff
@@ -106,6 +106,10 @@ Public Class sdgSummaries
         ucrReceiverSecondVariable.SetMeAsReceiver()
         ucrReceiverSecondVariable.SetIncludedDataTypes({"numeric"})
 
+        ucrReceiverOrderBy.Selector = ucrSelectorOrderBy
+        ucrReceiverOrderBy.SetMeAsReceiver()
+        ucrReceiverOrderBy.SetIncludedDataTypes({"numeric"})
```
Comment (dannyparsons) [inline instat/sdgSummaries.vb]: the `order_by` could be any type not just numeric

**Comment by Muthenya** [comment]
This is what we have now.

image

**Comment by rdstern** [comment]
Looks good to me

**Comment by dannyparsons** [comment]
Yep. And the selector/receiver hides when un checked

**Comment by Muthenya** [comment]
Danny, from your comment above, the Order by Checkbox should be disabled if none of the summaries are checked. Should I also implement this.

**Comment by dannyparsons** [comment]
Yes please

**Comment by Muthenya** [comment]
I have opened another pull request for this.

**Comment by shadrackkibet** [comment]
@Muthenya  thanks for taking over this.

**Comment by dannyparsons** [comment]
Since it contains all these commits it will be closed automatically when the other pull request is merged.

It's a good check to keep it open actually. If it doesn't close, it means it has changes not in the master.

**Comment by Muthenya** [comment]
Yes and I didn't want it to be closed now since it will close the issue it is fixing, which will now be done by the other pull request.

**Pull request merged**